design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,236 @@
+
+The `Column Selector` will display a list of all available columns as well as those already in use, if available. In addition, a complete list of all formulas will be available.
+
+The columns in `Column Selector` are listed in a hierarchical structure based on the links that exist in both directions between the base table and other tables. The icons indicate whether the linked records are a single or multiple record collection.
+
+Links are presented as expandable sections containing the columns from the linked tables.
+
+When adding input columns, the user can add them using the list controls or drop the columns directly into the result table.
+
+---
+Wireframes
+
+[Adding Columns](https://share.balsamiq.com/c/oyxTXxqSh8rLqU3JY71DWY.png)
+
+[Added Column](https://share.balsamiq.com/c/7U9yahZtYWX2G5obkyivoz.png)
+
+### 2.2 Input Column Sources
+
+Once the input columns have been added, the user can inspect them by selecting each from the result table or the list of columns in the query configuration panel, respectively. A section titled 'Source' will be included in the details for each column. The source column, table, and links will all be listed in this portion of the panel.
+
+The source will include `Links` only for data sourced via tables connected to or from the base table.
+
+---
+Wireframes
+
+[Column Sources](https://share.balsamiq.com/c/ewCxvkjS8VoScKfgb9hFos.png)
+
+### 2.3 Adding Formulas
+
+Formulas are used to generate new columns based on different parameters. To access the list of formulas, the user start the `Add Column` process and selects the option `From Formula` at the top of the inspector panel. Selecting a formula will open a form that users can fill out to determine the values of the new column.
+
+Depending on the selected formula, different settings will be available.
+More on formulas and specific details for each will be covered on a separate issue.
+
+Columns generated from formulas will display a formula icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Formula](https://share.balsamiq.com/c/vWfJ9sUYWJxed5Zo5WtPGi.png)
+
+### 2.4 Filtering Input Column Values
+
+Columns that link to multiple records can have filters applied to them to retrieve only values that match user-specified criteria. Multiple filters are allowed for each input column.
+
+Filter options will be determined by the data type of the input column.
+
+Filters can be added by clicking on the `Add Filter` option from the query configuration panel or directly from the results table column header menu.
+
+Columns with filters applied will display a filter icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Input Filter](https://share.balsamiq.com/c/tK2hZy6FB4zVYpN56ejpBk.png)
```

Comment:
The filter is applied on `actor.date_of_birth`, while the selected column is `actor.id`. If we add another column `actor.name`, would we show this applied filter in both columns?

Reply:
It would only affect the input column with the filter.